dev/tests/api-functional/framework/Magento/TestFramework/TestCase/WebapiAbstract.php

```diff
@@ -766,4 +766,60 @@ protected function assertWebApiCallErrors(array $serviceInfo, array $data, array
             }
         }
     }
+
+    /**
+     * Compare arrays recursively regardless of nesting.
+     * Can compare arrays that have both one level and n-level nesting.
+     * ```
+     *  [
+     * 'products' => [
+     *      'items' => [
+     *      [
+     *          'sku'       => 'bundle-product',
+     *          'type_id'   => 'bundle',
+     *          'items'     => [
+     *          [
+     *              'title'     => 'Bundle Product Items',
+     *              'sku'       => 'bundle-product',
+     *              'options'   => [
+     *              [
+     *                  'price' => 2.75,
+     *                  'label' => 'Simple Product',
+     *                  'product' => [
+     *                      'name'    => 'Simple Product',
+     *                      'sku'     => 'simple',
+     *                  ]
+     *              ]
+     *          ]
+     *      ]
+     *  ];
+     * ```
+     *
+     * @param array $expected
+     * @param array $actual
+     * @return array
+     */
+    public function compareArraysRecursively(array $expected, array $actual): array
```

Comment:
I like the idea but I would stand against feeding such a big monster as `WebapiAbstract.php` once again. Instead, I would suggest creating a separate class like `Magento\TestFramework\Helper\CompareArraysRecursively` with the only one public method (let's say `execute`) and use this class when necessary instead of the abstract methods.

@sivaschenko what do you think about this idea? 

Thank you!

Reply:
I agree with @rogyar.
Moreover, this functionality could be useful for other types of tests (e.g. Integration).

Reply:
Moved the function as suggested.